src/main/java/domain/Line.java

```diff
@@ -4,51 +4,58 @@
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
-import util.BridgeGenerator;
+import util.BooleanGenerator;
 
 public class Line {
     private final List<Bridge> bridges;
+    private final int lineSize;
 
     public Line(int personCount) {
-        this.bridges = IntStream.range(0,personCount - 1)
-                .mapToObj(it->Bridge.EMPTY)
+        this.bridges = IntStream.range(0, personCount - 1)
+                .mapToObj(it -> Bridge.EMPTY)
                 .collect(Collectors.toList());
+        this.lineSize = bridges.size();
     }
 
-    public void generate(BridgeGenerator bridgeGenerator) {
-        for (int bridgeIndex = 0; bridgeIndex < bridges.size(); bridgeIndex++) {
-            generateBridge(bridgeGenerator, bridgeIndex);
+    public void generate(BooleanGenerator booleanGenerator) {
+        for (int bridgeIndex = 0; bridgeIndex < lineSize; bridgeIndex++) {
+            generateBridge(booleanGenerator, bridgeIndex);
         }
     }
 
-    private void generateBridge(BridgeGenerator bridgeGenerator, int bridgeIndex) {
-        if (!hasBridgeInLeftOrRight(bridgeIndex, bridges.size() - 1)) {
-            bridges.set(bridgeIndex, bridgeGenerator.generate());
+    private void generateBridge(BooleanGenerator booleanGenerator, int bridgeIndex) {
+        Bridge randomBridge = generateRandomBridge(booleanGenerator);
+        if (!hasBridgeInLeftOrRight(bridgeIndex)) {
+            bridges.set(bridgeIndex, randomBridge);
         }
     }
 
-    private boolean hasBridgeInLeftOrRight(int bridgeIndex, int maxIndex) {
-        if (bridgeIndex == 0 && maxIndex == 0) {
-            return false;
-        }
-
-        if (bridgeIndex == 0) {
-            return hasBridgeInRight(bridgeIndex + 1);
+    public Bridge generateRandomBridge(BooleanGenerator booleanGenerator) {
+        if (booleanGenerator.generate()) {
+            return Bridge.EXIST;
         }
+        return Bridge.EMPTY;
+    }
```

Comment:
RandomBooleanGenerator 밖에 쓰이는 클래스가 없는데 
일어나지 않을일을 대비하여 BooleanGenerator를 만든것 같아요!
다른 요구사항이 생겨 RandomIntegerGenerator가 필요하면 지금 만든 인터페이스는 아무것도 쓰이지 않고 지워져버릴거같아요
다른 요구사항이 있을때 그때 공통적인 부분이 있다면 인터페이스를 만드는게 적절한 시점이라고 생각합니다 저는!

우선 인터페이스를 제거하고 generate() 메서드를 static 메서드로 변경하고`generateRandomBridge(RandomBooleanGenerator.generate())` 처럼 사용해보도록 변경해보는건 어떨까요?

```suggestion
    public Bridge generateRandomBridge(boolean randomBoolean) {
        if (randomBoolean) {
            return Bridge.EXIST;
        }
        return Bridge.EMPTY;
    }
```

테스트가 어려운건 LadderTest에서의 어려움일까요??
랜덤한 값으로 line이 생성되는것을 LineTest에서 진행했다면
LadderTest에서 굳이 TestGenerator까지 만들어서 테스트를 하는건 비효율적이라고 생각해요!
비효율적일뿐만 아니라 실제 구현에서 사용하지 않는 Generator로 테스트 하는건 큰 의미도 없는것 같아요
LineTest에서 true면 Bridge.Exist가 반환, false면 empty가 반환되는지만 확인하면 충분하지 않을까요?
가장 작은 단위의 테스트가 성공한다면 그 이후 상위 레벨에서는 흐름만 체크하는 테스트가 되면 될것 같습니다!

Reply:
1. 요구사항이 새로 생겼을 때 공통점에 맞게 확장하면 되기 때문에 현재 요구사항에서는 인터페이스가 필요없다.
2. 실제 구현에서 사용하지 않는 Generator로 테스트하는건 큰 의미가 없다.
라고 이해하면 될까요??

Reply:
TestGenerator를 만들었던 이유는
이번 미션 핵심인 `사다리 타기`를 Ladder의 findFinalPosition이 담당하고 있는데
사다리가 랜덤으로 생성되기 때문에, 사다리를 제대로 타는지(findFinalPosition이 제대로 작동하는지)를 확인하기 위해서
테스트코드에서는 원하는대로 사다리를 생성해야한다고 생각했습니다!
(핵심로직이기 때문에 실제 구현에서 사용하지 않는 Generator로라도 랜덤값을 제어하여 테스트를 해야한다고 생각했습니다)

말씀하신 것처럼 `true면 Bridge.Exist가 반환, false면 empty가 반환되는지`만 확인한다면,
핵심로직에 대한 테스트는 어떻게 진행할 수 있나요??

Reply:
```java
@DisplayName("사다리 게임")
    class climbLadder {
        @Test
        @DisplayName("사다리를 타고난 후의 위치값을 제대로 반환하는지 테스트")
        void climbTest() {
               /**
             *  0     1     2
             *  |-----|     |
             *  |     |-----|
             *  0     1     2
             */
            
            
            Line line = new Line(List.of(Bridge.EXIST, Bridge.EMPTY), 2);
            Line line2 = new Line(List.of(Bridge.EMPTY, Bridge.EXIST), 2);
            Ladder ladder = new Ladder(List.of(line, line2));


            //then
            assertThat(ladder.findFinalPosition(0)).isEqualTo(2);
            assertThat(ladder.findFinalPosition(1)).isEqualTo(0);
            assertThat(ladder.findFinalPosition(2)).isEqualTo(1);
        }
    }
```
말로 설명하기가 너무 어려워서 수도코드로 예시를 보여드립니다 참고만해주세요!
정적팩토리 메서드를 사용할 경우는 분명히 적절한 시점이 있을겁니다
제 기준에서는 이번 미션에서는 정적 팩토리 메서드가 오히려 깔끔한 코드를 작성하는데 방해가 되는것 같아요

Reply:
Line의 lineSize 필드도 굳이 필요없을것 같아요
birdges.size()를 활용해도 되지 않을까요?